core/src/network_manager/manager/control_directives.rs

```diff
@@ -0,0 +1,192 @@
+
+                // Spawn a separate task to asynchronously dial the peer and respond to
+                // the handshake manager's brokerage request
+                let allow_local = self.allow_local;
+                let handshake_queue_clone = self.handshake_work_queue.clone();
+                tokio::spawn(async move {
+                    if let Err(e) = Self::broker_mpc_net(
+                        allow_local,
+                        request_id,
+                        peer_port,
+                        local_port,
+                        local_role,
+                        known_peer_addrs,
+                        handshake_queue_clone,
+                    )
+                    .await
+                    {
+                        log::error!("error brokering MPC network: {e}");
+                    }
+                });
+
+                Ok(())
+            }
+
+            // Inform the network manager that the gossip server has warmed up the local node in
+            // the cluster by advertising the local node's presence
+            //
+            // The network manager delays sending pubsub events until the gossip protocol has warmed
+            // up, because at startup, there are no known peers to publish to. The network manager gives
+            // the gossip server some time to discover new addresses before publishing to the network.
+            ManagerControlDirective::GossipWarmupComplete => {
+                self.warmup_finished = true;
+                // Forward all buffered messages to the network
+                for buffered_message in self.warmup_buffer.drain(..).collect_vec() {
+                    self.forward_outbound_pubsub(buffered_message.topic, buffered_message.message)?;
+                }
+
+                Ok(())
+            }
+        }
+    }
+
+    /// Broker an MPC net between the local and remote peer using the given known addresses
+    async fn broker_mpc_net(
+        allow_local: bool,
+        request_id: Uuid,
+        peer_port: u16,
+        mut local_port: u16,
+        local_role: ConnectionRole,
+        known_peer_addrs: Vec<Multiaddr>,
+        handshake_work_queue: TokioSender<HandshakeExecutionJob>,
+    ) -> Result<(), NetworkManagerError> {
+        // Connect on a side-channel to the peer
+        let party_id = local_role.get_party_id();
+
+        let mpc_net = match local_role {
+            ConnectionRole::Dialer => {
+                // Attempt to dial the peer on all known addresses
+                let mut brokered_net = None;
+                for peer_addr in known_peer_addrs
+                    .into_iter()
+                    .filter_map(|addr| multiaddr_to_socketaddr(&addr, peer_port))
+                    .filter(|addr| is_dialable_addr(addr, allow_local))
+                {
+                    let local_addr: SocketAddr =
+                        format!("127.0.0.1:{:?}", local_port).parse().unwrap();
```

Comment:
maybe a bit out-of-scope, but just want to confirm that this is still the right way to get the local address after the bug fix last week regarding fetching network interfaces?
same goes for the `local_addr` and `peer_addr` in the `ConnectionRole::Listener` match arm below

Reply:
The local addr does not matter for the `Dialer` and the peer_addr does not matter for the `Listener`, I'll change it though so it is clearer to read